community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java

```diff
@@ -73,90 +77,147 @@
         neoStores.getMetaDataStore().setLastCommittedAndClosedTransactionId(
                 initialIds.lastCommittedTransactionId(), initialIds.lastCommittedTransactionChecksum(),
                 BASE_TX_COMMIT_TIMESTAMP, initialIds.lastCommittedTransactionLogByteOffset(),
                 initialIds.lastCommittedTransactionLogVersion() );
-        this.propertyKeyRepository = new BatchingPropertyKeyTokenRepository(
-                neoStores.getPropertyKeyTokenStore() );
-        this.labelRepository = new BatchingLabelTokenRepository(
-                neoStores.getLabelTokenStore() );
-        this.relationshipTypeRepository = new BatchingRelationshipTypeTokenRepository(
-                neoStores.getRelationshipTypeTokenStore() );
+    }
+
+    /**
+     * Called when expecting a previous attempt/state of a database to open, where some store files should be kept,
+     * but others deleted. All temporary stores will be deleted in this call.
+     *
+     * @param mainStoresToKeep {@link Predicate} controlling which files to keep, i.e. {@code true} means keep, {@code false} means delete.
+     * @param tempStoresToKeep {@link Predicate} controlling which files to keep, i.e. {@code true} means keep, {@code false} means delete.
+     * @throws IOException on I/O error.
+     */
+    public void pruneAndOpenExistingStore( Predicate<StoreType> mainStoresToKeep, Predicate<StoreType> tempStoresToKeep ) throws IOException
+    {
+        deleteStoreFiles( TEMP_NEOSTORE_NAME, tempStoresToKeep );
+        deleteStoreFiles( DEFAULT_NAME, mainStoresToKeep );
+        instantiateStores();
+    }
 
-        // Instantiate the temporary stores
-        temporaryNeoStores = newStoreFactory(
-                "temp." + DEFAULT_NAME, DELETE_ON_CLOSE ).openNeoStores( true, RELATIONSHIP_GROUP );
+    private void deleteStoreFiles( String storeName, Predicate<StoreType> storesToKeep )
+    {
+        FileSystemAbstraction fs = pageCache.getCachedFileSystem();
+        for ( StoreType type : StoreType.values() )
+        {
+            if ( type.isRecordStore() && !storesToKeep.test( type ) )
+            {
+                for ( StoreFileType fileType : StoreFileType.values() )
+                {
+                    fs.deleteFile( new File( storeDir, fileType.augment( storeName + type.getStoreFile().fileNamePart() ) ) );
+                }
+            }
+        }
+    }
 
-        // Initialize kernel extensions
+    private void instantiateKernelExtensions()
+    {
+        life = new LifeSupport();
         life.start();
         labelScanStore = new NativeLabelScanStore( pageCache, storeDir, FullStoreChangeStream.EMPTY, false, new Monitors(),
                 RecoveryCleanupWorkCollector.IMMEDIATE );
         life.add( labelScanStore );
     }
 
+    private void instantiateStores() throws IOException
+    {
+        neoStores = newStoreFactory( DEFAULT_NAME ).openAllNeoStores( true );
+        // TODO why do we need this counts store thing here?
```

Comment:
TODO

Reply:
A mystery I haven't quite understood yet. Can it stay so that some other adventurer or future-me will find out?

Reply:
)) ok